docker-compose-non-dev.yml

```diff
@@ -14,7 +14,6 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
-x-superset-image: &superset-image apachesuperset.docker.scarf.sh/apache/superset:${TAG:-latest}
```

Comment:
I personally think it's kind of cool to have `non-dev` point to a pre built image TAG, also this docker-compose does not mount current code into the container like `docker-compose.yaml` does, so non deterministic cases probably do not apply on this case.

Reply:
About this, I think both use cases are valid. To me if I'm in a repo and on a specific ref (a branch, a release tag, or my own little branch with a feature), and I run some docker-related thing (whether it's `docker build` or a `docker-compose` related thing) I'm assuming that what I'm building is the particular ref I'm into right now. 

I think the 2 options I want to provide here are really just "interactive" where we mount the code, and "non-interactive" where it's just immutable set of dockers that get me a fully working testable cluster that is lined up with the branch.

Now maybe we should ADD a new way to do `docker-compose-any-image.yml` that would work along with a TAG env var.

Reply:
fine by me!
But `docker-compose-non-dev.yaml` will not mount any code into the containers, so it does not matter if your on a branch or tag, it will just use `latest` or `TAG`, this is prob a good candidate for `docker-compose-any-image.yml`, no?

Reply:
ok, I'm making a bunch of changes here and re-writing the docs too...